### PR TITLE
Use smaller organized namespaces.

### PR DESCRIPTION
**Pull request by exaexa**
This will eventually pick out all stuff from #672 .

Current status:
- some basic stuff works (io, types, logging), gives a pretty good precedens for the rest
- tests: TODO
- docs: absolute TODO, will do last, but probably doable with `sed`

List of random stuff I should remember to check before merging:
- [x] also move the test files (squash `/base/`)
- [x] all internal modules have docs
- [x] tabbing works

postponed to #673 :
- [ ] docs coverage is full
- [ ] ugly stuff is hidden into the specialized `Internal` modules (preferably most of the `_underscored` hidden things that we currently have)

### CODE REVIEW

File: src/COBREXA.jl [hunk 1]
```diff
@@ -39,32 +39,73 @@ using Serialization
...
+            sym in [Symbol(@__MODULE__), :eval, :include] && continue
+            startswith(string(sym), ['_', '#']) && continue
+            sym == :Internal && continue
+            @eval export $(Expr(:$, :sym))
+        end
+    end
+end
+
+@export_locals
+end
```
Comment (stelmo) [inline src/COBREXA.jl:83]: This is so good

File: src/types.jl [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
+module Types
+using ..ModuleTools
+@dse
```
Comment (stelmo) [inline src/types.jl:4]: so much better

**Review by stelmo — APPROVED** [review]
Love it

**Comment by stelmo** [comment]
This is truly amazing

**Review by stelmo — APPROVED** [review]
This is amazing

File: src/modules.jl [hunk 3]
```diff
@@ -0,0 +1,34 @@
+module ModuleTools
```
[the remark this answers is not in the document]
Comment (exaexa) [reply src/modules.jl:1]: true

**Comment by exaexa** [comment]
I'll finish it ASAP this week together with several other things

**Comment by exaexa** [comment]
the 2 rest issues are left for `next`

**Pull request merged**